File: backend/src/main/java/com/equisoft/dca/backend/project/model/Compiler.java
package com.equisoft.dca.backend.project.model;

import com.equisoft.dca.backend.common.model.BaseEnum;

import lombok.Getter;

@Getter
public enum Compiler implements BaseEnum<Integer> {
	GENERIC(1, "Generic"),
	MICROFOCUS(2, "MicroFocus"),
	FUJITSU(3, "Fujitsu"),
	VISUAL_AGE(4, "Visual Age"),
	COBOL_II(5, "COBOL II"),
	DOUBLE_BYTE(6, "Double Byte");

	private final int id;

	private final String name;

	Compiler(int id, String name) {
		this.id = id;
		this.name = name;
	}

	@Override
	public Integer getId() {
		return id;
	}
}

File: react/src/lib/domain/entities/Compiler.ts
import BaseEnum from './BaseEnum';

export default class Compiler extends BaseEnum {

    public static GENERIC = new Compiler('1', "project:compilerEnum:generic");
    public static MICROFOCUS = new Compiler('2', "project:compilerEnum:microFocus");
    public static FUJITSU = new Compiler('3', "project:compilerEnum:fujitsu");
    public static VISUAL_AGE = new Compiler('4', "project:compilerEnum:visualAge");
    public static COBOL_II = new Compiler('5', "project:compilerEnum:cobolII");
    public static DOUBLE_BYTE = new Compiler('6', "project:compilerEnum:doubleByte");

    public constructor(id?: string, name?: string) {
        super(id, name);
    }
}
